code/sorting/insertion_sort/insertion_sort.java

```diff
@@ -1,33 +1,21 @@
 // Part of Cosmos by OpenGenus Foundation
 class InsertionSort {
     void sort(int arr[]) {
-        for (int i = 1; i < arr.length; ++i) {
-            int key = arr[i];	// current element is the key
-            int j = i - 1;
-            while (j >= 0 && arr[j] > key) {	// Move elements of arr[0..i-1] greater than key,
-                arr[j + 1] = arr[j];			// to one position ahead of their current position
-                j = j - 1;
+        for (int i = 0; i < arr.length; ++i) {
+            for (int j = i; j > 0 && arr[j - 1] > arr[j]; j--) {
+                int temp = arr[j];
+                arr[j] = arr[j - 1];
+                arr[j - 1] = temp;
             }
-            arr[j+1] = key;
         }
+        System.out.println("After:" + Arrays.toString(arr));
```

Comment:
Move this outside the class. A sorting function should **only** sort.

Reply:
Thank you for your comment.